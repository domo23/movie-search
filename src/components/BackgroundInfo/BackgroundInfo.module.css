.main {
    display: flex;
    width: 100%;
    height: 100%;
    justify-content: center;
    align-items: center;
    color: gray;
    font-size: 2rem;
    padding: 20px;
    box-sizing: border-box;
}