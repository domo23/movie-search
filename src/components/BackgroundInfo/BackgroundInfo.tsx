import styles from './BackgroundInfo.module.css'

export type Props = {
    msg: string
}

function BackgroundInfo({msg}: Props) {
    return (
        <div className={styles.main} data-testid="background-info">{msg}</div>
    )
}

export default BackgroundInfo;