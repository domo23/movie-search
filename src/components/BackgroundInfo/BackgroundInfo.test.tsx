import { fireEvent, render } from "@testing-library/react";
import BackgroundInfo from "./BackgroundInfo";
import { Props } from "./BackgroundInfo";

function renderBackgroundInfo(props: Partial<Props> = {}) {
  const defaultProps: Props = {
    msg: "",
  };
  return render(<BackgroundInfo {...defaultProps} {...props} />);
}

it("render background msg, snapshot test", () => {
  const { getByTestId } = renderBackgroundInfo();
  const element = getByTestId("background-info");
  expect(element).toMatchSnapshot();
});

it("test background msg", async () => {
  const msg = 'test string';
  const { getByTestId } = renderBackgroundInfo({ msg });
  const element = getByTestId("background-info");
  
  expect(element).toHaveTextContent(msg);
});
