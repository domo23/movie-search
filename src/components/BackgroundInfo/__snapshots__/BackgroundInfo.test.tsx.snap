// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render background msg, snapshot test 1`] = `
<div
  class="main"
  data-testid="background-info"
/>
`;
